### ISSUE TITLE
PetaJakarta is not working anymore

### ISSUE THREAD

**Issue by timlinux** [inasafe/inasafe#3752]
### Problem

In our Jan 2017 site visit we found out that PetaJakarta is moving to PetaBencana but the old API still seemed to work for fetching data. Now it seems to just return nothing....

### Expected Behaviour

Working PetaJakarta tool...

### Environment
* InaSAFE : 
* QGIS : 
* OS : 
* Data used :
  * Exposure : 
  * Hazard : 
  * Aggregation :

### Proposed Solution

@timlinux write to Tim from PJ and find out what the deal is. Consider removing this tool from 4.0 if there is no prospect of it working.

**Comment by Charlotte-Morgan** [comment]
@timlinux - apologies for the delay getting you this info.
Here are contact details for the former PetaJakarta principle investigators:
Etienne e-mail: turpin@mit.edu or etienne@anexact.org
Tomas: tomasholderness@gmail.com

I hope you have success with these contacts; if not I am happy to follow up with UoW to find out if they are able to help.

**Comment by lucernae** [comment]
The REST API now:

```
https://data.petabencana.id/floods?city=jbd&geoformat=geojson&format=json&minimum_state=1
```

It is returned as REST result (not as GeoJSON directly)

```
{"statusCode":200,"result":{"type":"FeatureCollection","features":[]}}
```

So we need to get the value in the "result" mapping.

Also they provided a testing server with fixed data:

```
https://data-dev.petabencana.id/floods?city=jbd&geoformat=geojson&format=json&minimum_state=1
```

**Comment by timlinux** [comment]
Thanks @Charlotte-Morgan 

Just for follow up I did write to Tom who has replied with some hints. @lucernae is now looking into it. @myarjunar will you please make sure the desktop tool works too once @lucernae has figured out the procedure to follow?

Thanks

**Comment by lucernae** [comment]
Ah, I already made the fix since it is straightforward. @myarjunar you can review and take over from there.

**Comment by lucernae** [comment]
@myarjunar can you take over from here?
We need to refactor all PetaJakarta, petajakarta, Peta Jakarta references to PetaBencana in the code.

**Comment by myarjunar** [comment]
@lucernae ok sip :+1: , I will take care of this issue

**Comment by lucernae** [comment]
Thank's @myarjunar I merged the PR. You can start another one :)

**Comment by Gustry** [comment]
@myarjunar Can we close this one too ?

**Comment by timlinux** [comment]
I would like to add one last small tweak request: If it returns zero rows, don't add a layer to QGIS but rather use e.g. a QGIS status message to tell the users 'there are no floods at this time'. An empty flood layer is not very useful in the legend.

**Comment by Gustry** [comment]
PetaBencana has 3 cities in the API : https://data.petabencana.id/cities
* jbd for Jabodetabek
* bdg for Bandung
* sby for Surabaya

Can't we add a combobox to support Bandung and Surabaya ?

**Comment by timlinux** [comment]
That would be cool.

**Comment by Gustry** [comment]
You can go ahead @myarjunar ;-)

**Comment by myarjunar** [comment]
Ok then! The truth has been spoken

**Comment by Charlotte-Morgan** [comment]
Point to note here.  Is there any data? I think this is ok for the download tool though it may not work if there are no data yet. I am also reluctant without metadata on the API - we know very little about the data nor if it is validated. Please do not change flood realtime as this is a product negotiated with partners in Jakarta and DMI has no plan to extend realtime flood to other cities.

**Comment by lucernae** [comment]
@Charlotte-Morgan realtime will not be affected on this repo. I will keep note that flood realtime will only use API for JBD only, (but we have to update the API).

For the data, just some suggestion, maybe we should provide a way to switch the url to PetaBencana's dev url? The data is dummy though, but it might be useful to check the keywords/styles and see what it will look like without having to wait for real flood.

**Comment by Gustry** [comment]
@myarjunar I think make a file in definitions for petabencana with these cities and these two URLs.

As it's dummy data, we will not switch often to dev API, so I think we can keep it simple and switch easily by changing a variable in is enough, no ? If you want a checkbox, display it only if dev mode is on.

**Comment by Charlotte-Morgan** [comment]
supporting anythin but Jakarta is a low priority for now and can wait

**Comment by myarjunar** [comment]
closed in #3866
